client/status/common.go

```diff
@@ -174,3 +174,48 @@ func (x *SignatureVerification) SetMessage(v string) {
 func (x SignatureVerification) Message() string {
 	return x.v2.Message()
 }
+
+// NodeUnderMaintenance describes failure status for nodes being under maintenance.
+// Instances provide Status and StatusV2 interfaces.
+type NodeUnderMaintenance struct {
+	v2 status.Status
+}
+
+// Error implements the error interface.
+func (x NodeUnderMaintenance) Error() string {
+	return errMessageStatusV2(
+		globalizeCodeV2(status.NodeUnderMaintenance, status.GlobalizeCommonFail),
+		x.v2.Message(),
+	)
+}
+
+func (x *NodeUnderMaintenance) fromStatusV2(st *status.Status) {
+	x.v2 = *st
+}
+
+// ToStatusV2 implements StatusV2 interface method.
+// If the value was returned by FromStatusV2, returns the source message.
+// Otherwise, returns message with
+//   - code: NODE_UNDER_MAINTENANCE;
+//   - string message: written message via SetMessage;
+//   - details: empty.
+func (x NodeUnderMaintenance) ToStatusV2() *status.Status {
+	x.v2.SetCode(globalizeCodeV2(status.NodeUnderMaintenance, status.GlobalizeCommonFail))
+	return &x.v2
+}
+
+// SetMessage writes signature verification failure message.
+// Message should be used for debug purposes only.
+//
+// See also Message.
+func (x *NodeUnderMaintenance) SetMessage(v string) {
```

Comment:
So the message will show what kind of repair work node is currently performing, right? I'm trying to realize is this really needed.

Reply:
Yes, and in this case it should be useful, because there are a lot of possible reasons for this state. For example, we can tell what moment in time we are expected to go ONLINE.